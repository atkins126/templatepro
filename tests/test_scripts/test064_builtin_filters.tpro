{{:|version}}
{{:|uppercase,"pippo"}}
{{:|lowercase,"PIPPO"}}

{{:value2}} => {{:value2|lowercase}}
{{:valuedate|formatdatetime,"yyyy-mm-dd"}}

filter: datetostr
1) {{:valuedate|datetostr}}
2) {{:valuedate|datetostr,"yy-mm-dd"}}


filter: datetimetostr
1) {{:valuedatetime|datetimetostr}}
2) {{:valuedatetime|datetimetostr,"dddd d mmmm yyyy hh:nn:ss"}}


{{for obj in objects}}
  {{:obj.Prop1 | lpad , 10 }}
{{endfor}}

empty var ==> {{:datevalue|datetostr}}

{{if intvalue1|gt,0}}
gt 1 > 0
{{endif}}

{{if intvalue0|lt,123}}
lt 0 < 123
{{endif}}

{{if intvalue1|ge,0}}
ge 1 > 0
{{endif}}

{{if intvalue0|le,0}}
le 0 = 0
{{endif}}

{{if intvalue0|eq,0}}
eq 0 = 0
{{endif}}

{{if intvalue0|ne,1}}
ne 0 = 1
{{endif}}


{{for cust in customers}}
  Customer uppercase: {{:cust.Name|uppercase}}
  Customer lowercase: {{:cust.Name|lowercase}}  
  Customer lpad: {{:cust.Name|lpad,10}}
  Customer lpad: {{:cust.Name|lpad,10,"_"}}
  Customer rpad: {{:cust.Name|rpad,10}}
  Customer rpad: {{:cust.Name|rpad,10,"_"}} 
  Customer capitalize: {{:cust.Name|capitalize}}   
  Customer contains "a": {{:cust.Name|contains,"a"}} 
  Customer contains_ignore_case "a": {{:cust.Name|contains_ignore_case,"a"}} 

{{endfor}}

