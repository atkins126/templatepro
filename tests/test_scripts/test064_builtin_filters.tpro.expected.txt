0.7.1
PIPPO
pippo

DANIELE2 => daniele2
2024-08-20

filter: datetostr
1) 2024-08-20
2) 24-08-20


filter: datetimetostr
1) 2024-08-20 10:20:30
2) marted&igrave; 20 agosto 2024 10:20:30


    value1.1
    value1.2
    value1.3

empty var ==> 

gt 1 > 0

lt 0 < 123

ge 1 > 0

le 0 = 0

eq 0 = 0

ne 0 = 1


  Customer uppercase: FORD
  Customer lowercase: ford  
  Customer lpad:       Ford
  Customer lpad: ______Ford
  Customer rpad: Ford      
  Customer rpad: Ford______ 
  Customer capitalize: Ford   
  Customer contains "a": False 
  Customer contains_ignore_case "a": False 

  Customer uppercase: FERRARI
  Customer lowercase: ferrari  
  Customer lpad:    Ferrari
  Customer lpad: ___Ferrari
  Customer rpad: Ferrari   
  Customer rpad: Ferrari___ 
  Customer capitalize: Ferrari   
  Customer contains "a": True 
  Customer contains_ignore_case "a": True 

  Customer uppercase: LOTUS
  Customer lowercase: lotus  
  Customer lpad:      Lotus
  Customer lpad: _____Lotus
  Customer rpad: Lotus     
  Customer rpad: Lotus_____ 
  Customer capitalize: Lotus   
  Customer contains "a": False 
  Customer contains_ignore_case "a": False 

  Customer uppercase: FCA
  Customer lowercase: fca  
  Customer lpad:        FCA
  Customer lpad: _______FCA
  Customer rpad: FCA       
  Customer rpad: FCA_______ 
  Customer capitalize: Fca   
  Customer contains "a": False 
  Customer contains_ignore_case "a": True 

  Customer uppercase: HYUNDAI
  Customer lowercase: hyundai  
  Customer lpad:    Hyundai
  Customer lpad: ___Hyundai
  Customer rpad: Hyundai   
  Customer rpad: Hyundai___ 
  Customer capitalize: Hyundai   
  Customer contains "a": True 
  Customer contains_ignore_case "a": True 

  Customer uppercase: DE TOMASO
  Customer lowercase: de tomaso  
  Customer lpad:  De Tomaso
  Customer lpad: _De Tomaso
  Customer rpad: De Tomaso 
  Customer rpad: De Tomaso_ 
  Customer capitalize: De Tomaso   
  Customer contains "a": True 
  Customer contains_ignore_case "a": True 

  Customer uppercase: DODGE
  Customer lowercase: dodge  
  Customer lpad:      Dodge
  Customer lpad: _____Dodge
  Customer rpad: Dodge     
  Customer rpad: Dodge_____ 
  Customer capitalize: Dodge   
  Customer contains "a": False 
  Customer contains_ignore_case "a": False 

  Customer uppercase: TESLA
  Customer lowercase: tesla  
  Customer lpad:      Tesla
  Customer lpad: _____Tesla
  Customer rpad: Tesla     
  Customer rpad: Tesla_____ 
  Customer capitalize: Tesla   
  Customer contains "a": True 
  Customer contains_ignore_case "a": True 

  Customer uppercase: KIA
  Customer lowercase: kia  
  Customer lpad:        Kia
  Customer lpad: _______Kia
  Customer rpad: Kia       
  Customer rpad: Kia_______ 
  Customer capitalize: Kia   
  Customer contains "a": True 
  Customer contains_ignore_case "a": True 

  Customer uppercase: TATA
  Customer lowercase: tata  
  Customer lpad:       Tata
  Customer lpad: ______Tata
  Customer rpad: Tata      
  Customer rpad: Tata______ 
  Customer capitalize: Tata   
  Customer contains "a": True 
  Customer contains_ignore_case "a": True 

  Customer uppercase: VOLKSWAGEN
  Customer lowercase: volkswagen  
  Customer lpad: Volkswagen
  Customer lpad: Volkswagen
  Customer rpad: Volkswagen
  Customer rpad: Volkswagen 
  Customer capitalize: Volkswagen   
  Customer contains "a": True 
  Customer contains_ignore_case "a": True 

  Customer uppercase: AUDI
  Customer lowercase: audi  
  Customer lpad:       Audi
  Customer lpad: ______Audi
  Customer rpad: Audi      
  Customer rpad: Audi______ 
  Customer capitalize: Audi   
  Customer contains "a": False 
  Customer contains_ignore_case "a": True 

  Customer uppercase: SKODA
  Customer lowercase: skoda  
  Customer lpad:      Skoda
  Customer lpad: _____Skoda
  Customer rpad: Skoda     
  Customer rpad: Skoda_____ 
  Customer capitalize: Skoda   
  Customer contains "a": True 
  Customer contains_ignore_case "a": True 


